documentation/modules/ROOT/pages/integrations/outbox.adoc

```diff
@@ -133,35 +133,190 @@ The table name to be used when creating the outbox table.
+|boolean
+|false
+
+|`quarkus.debezium-outbox.aggregate-type-name`::
+The column name for the event aggregate type column.
+|string
+|`aggregatetype`
+
+|`quarkus.debezium-outbox.aggregate-type.column-definition`::
+The database-specific column definition for the aggregate type.
+|string
+|
+
+|`quarkus.debezium-outbox.aggregate-type.length`::
+The length of the event aggregate type column.
+|integer
+|
+
+|`quarkus.debezium-outbox.aggregate-type.scale`::
+The scale of the event aggregate type column.
+|integer
+|
+
+|`quarkus.debezium-outbox.aggregate-type.precision`::
+The precision of the event aggregate type column.
+|integer
+|
+
+|`quarkus.debezium-outbox.aggregate-type.converter`::
+The JPA AttributeConverter for the event aggregate type column.
+|string
+|
+
+|`quarkus.debezium-outbox.aggregate-type.nullable`::
+Whether the event aggregate type column is nullable.
+|boolean
+|false
+
+|`quarkus.debezium-outbox.type.name`::
 The column name for the event type column.
 |string
 |`type`
 
-|`quarkus.debezium-outbox.timestamp-column-name`::
+|`quarkus.debezium-outbox.type.column-definition`::
+The database-specific column definition for the event type.
+|string
+|
+
+|`quarkus.debezium-outbox.type.length`::
+The length of the event type column.
+|integer
+|
+
+|`quarkus.debezium-outbox.type.scale`::
+The scale of the event type column.
+|integer
+|
+
+|`quarkus.debezium-outbox.type.precision`::
+The precision of the event type column.
+|integer
+|
+
+|`quarkus.debezium-outbox.type.converter`::
+The JPA AttributeConverter for the event type column.
+|string
+|
+
+|`quarkus.debezium-outbox.type.nullable`::
+Whether the event type column is nullable.
+|boolean
+|false
+
+|`quarkus.debezium-outbox.timestamp.name`::
 The column name for the event timestamp column.
 |string
 |`timestamp`
 
-|`quarkus.debezium-outbox.payload-column-name`::
+|`quarkus.debezium-outbox.timestamp.column-definition`::
+The database-specific column definition for the event timestamp.
+|string
+|
+
+|`quarkus.debezium-outbox.timestamp.length`::
+The length of the event timestamp column.
+|integer
+|
+
+|`quarkus.debezium-outbox.timestamp.scale`::
+The scale of the event timestamp column.
+|integer
+|
+
+|`quarkus.debezium-outbox.timestamp.precision`::
+The precision of the event timestamp column.
+|integer
+|
+
+|`quarkus.debezium-outbox.timestamp.converter`::
+The JPA AttributeConverter for the event timestamp column.
+|string
+|
+
+|`quarkus.debezium-outbox.timestamp.nullable`::
+Whether the event timestamp column is nullable.
+|boolean
+|false
+
+|`quarkus.debezium-outbox.payload.name`::
 The column name for the event payload column.
 |string
 |`payload`
 
-|`quarkus.debezium-outbox.aggregate-type-column-name`::
-The column name for the event aggregate type column.
+|`quarkus.debezium-outbox.payload.column-definition`::
+The database-specific column definition for the event payload.
 |string
-|`aggregatetype`
+|
+
+|`quarkus.debezium-outbox.payload.length`::
```

Comment:
Would it perhaps make sense to simply have a single property (besides the attribute converter) that takes the plain column type which gets passed through:

```
quarkus.debezium-outbox.payload.column.definition=NOT NULL VARCHAR(2000)
```

WDYT?

Reply:
We certainly could do that; would this apply to all columns by simply removing the length, scale, and precision configurable options?

Reply:
I went ahead and removed the length/scale/precision/nullable options in favor of just using the column-definition option for now.